pkg/helm/helm.go

```diff
@@ -308,14 +347,31 @@ func (h *Helm) LoadChart(ref *LoadChartConfig) (c *chart.Chart, err error) {
 	if err != nil {
 		return
 	}
+
 	log.Printf("Using chart path %v", chartPath)
+
 	// Check chart requirements to make sure all dependencies are present in /charts
-	if c, err = chartutil.Load(chartPath); err != nil {
+	if c, err = loadChartWithLockfile(chartPath, ref.LockFile); err != nil {
 		return
 	}
 
-	req, e := chartutil.LoadRequirements(c)
-	if e == nil {
+	lock, err := chartutil.LoadRequirementsLock(c)
+	if err != nil && err != chartutil.ErrLockfileNotFound {
+		return nil, err
+	}
+
+	req, err := chartutil.LoadRequirements(c)
+	if err != nil && err != chartutil.ErrRequirementsNotFound {
+		return nil, err
```

Comment:
this early exit results in requirements.lock not being checked if requirements.yaml is empty

Reply:
Hmm... `requirements.lock` is just a digest for the data in `requirements.yaml`. I don't know that I've ever encountered a repo with an empty `requirements.yaml` but a `requirements.lock`. Can you provide an example?

I'm also getting to think that `requirements.lock` isn't the guarantee you're looking for.

Are you familiar with Helm provenance files? They are more or less a signed checksum of the chart's `.tgz`. This would verify that the contents of the chart itself have not been changed, but comes with certain limitations:

1. The chart must be packed. You can't verify an unpacked chart.
2. The provenance file must be served from the same location as the packed chart. The logic in Helm if you request chart verification just appends `.prov` to the URL to the packed chart.

So that would mean that, for you to author your own provenance files, you'd need to run your own chart repo.

If this covers your use-case, there's actually no change needed to the plugin in order to facilitate this. If you're going down this path, you could also use `file://` references in your `requirements.yaml` and vendor your chart dependencies in your packed chart. At that point, your provenance also accounts for the actual checksums of your dependencies, rather than `requirements.lock` simply ensuring that exact dependency version numbers are re-used across builds.

Reply:
We could also try to come up with our own  checksum logic that would work with existing public charts having non-vendored dependencies (if requirements.lock exists or is injected a la this PR), but it would be an entirely new construct.

We'd need tooling to calculate these checksums, but then a chart inflator could specify an expected checksum and you wouldnt need your own repo, customized charts, keyring, etc.

Is that more or less what you'd want?

Reply:
> I don't know that I've ever encountered a chart with an empty `requirements.yaml` but a `requirements.lock`. Can you provide an example?

I was trying to lock a reference to a remote chart with no dependencies (cert-manager).
Might be worth adding a test case for it?



> Are you familiar with Helm provenance files? They are more or less a signed checksum of the chart's `.tgz`. This would verify that the contents of the chart itself have not been changed, but comes with certain limitations:

For our usecase I don't really care for *who* wrote the helm chart; just that it hasn't been changed out from underneath us maliciously.